src/Compilers/VisualBasic/Test/Emit/ExpressionTrees/CodeGenExprLambda.vb

```diff
@@ -2807,6 +2807,195 @@ Lambda(
 ]]>)
         End Sub
 
+        <Fact()>
+        Public Sub Relaxation01()
+            Dim file = <file name="expr.vb"><![CDATA[
+Imports System
+Imports System.Linq.Expressions
+Imports System.Reflection
+
+Module Module1
+
+    Sub Main()
+        Dim o As New C1(Of C0)
+    End Sub
+
+End Module
+
+Class C0
+    Public Function Process() As Boolean
+        Return False
+    End Function
+
+    Public Sub ProcessSub()
+    End Sub
+End Class
+
+Class C1(Of T As C0)
+    Public ProcessMethodSub As MethodInfo = RegisterMethod(Sub(c) c.ProcessSub())
+    Public ProcessMethod As MethodInfo = RegisterMethod(Function(c) c.Process)
+
+    Public Function RegisterMethod(methodLambdaExpression As Expression(Of Action(Of T))) As MethodInfo
+        System.Console.WriteLine(methodLambdaExpression.ToString())
+
+        Return Nothing
+    End Function
+
+End Class
+
+
+]]></file>
+
+            TestExpressionTrees(file,
+            <![CDATA[
+c => c.ProcessSub()
+c => c.Process()
```

Comment:
Perhaps, but I think we need to understand all the implications of this change to make the call. Because if the tree is not compilable/executable it might be appropriate to take another look at whether we want to restore the compatibility, because current Roslyn behavior could be seen as a bug fix.